module.nix

```
@@ -0,0 +1,112 @@
+      description = "Port of the server (will be passed using --bind flag)";
+    };
+    address = lib.mkOption {
+      type = lib.types.str;
+      default = "127.0.0.1";
+      description = "The address the server will listen on (will be passed using --host flag)";
+    };
+    flags = lib.mkOption {
+      type = with lib.types; listOf str;
+      default = [ ];
+      description = "Flags to pass to the server command-line";
+    };
+    settings = lib.mkOption {
+      type = with lib.types; submodule {
+        freeformType = attrsOf str;
+        options.SQLALCHEMY_DATABASE_URL = lib.mkOption {
+          description = "Database url";
+        };
+      };
+      description = "Settings to pass as environment variables";
+    };
+    envFile = lib.mkOption {
+      type = with lib.types; nullOr path;
+      description = "Environment file to append to settings.";
+      default = null;
+    };
+    domain = lib.mkOption {
+      type = lib.types.str;
+      description = "Hostname for reverse proxy config. To configure";
+    };
+    nginx = lib.mkOption {
+      type = lib.types.submodule (
+        lib.recursiveUpdate
+          (import (modulesPath + "/services/web-servers/nginx/vhost-options.nix") { inherit config lib; }) {}
+      );
+      default = {};
+      example = lib.literalExpression ''
+        {
+          serverAliases = [
+            "arkheon.''${config.networking.domain}"
+          ];
+          # To enable encryption and let let's encrypt take care of certificate
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+      description = lib.mdDoc ''
+        With this option, you can customize an nginx virtual host which already
+        has sensible defaults for Arkheon. If enabled, then by default, the
+        serverName is ''${domain}, If this is set to null, no nginx virtualHost
+        will be configured.
+      '';
+    };
+  };
+  config = {
+    services.arkheon.flags =
+      lib.optional (cfg.address != null) "--host ${cfg.address}" ++
+      lib.optional (cfg.port != null) "--port ${builtins.toString cfg.port}";
+    systemd.services.arkheon = lib.mkIf cfg.enable {
+      environment = cfg.settings;
+      path = [ cfg.pythonEnv ];
+      script = lib.concatStringsSep " " ([ cfg.command "arkheon:app" ] ++ cfg.flags);
+      serviceConfig = {
+        StateDirectory = "arkheon";
+        DynamicUser = true;
+        EnvironmentFile = [ (lib.optional (cfg.envFile != null) cfg.envFile) ];
+      };
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+    };
+    services.nginx = lib.mkIf (cfg.nginx != null) {
+      enable = true;
+      virtualHosts.${cfg.domain} = {
+        locations = {
+          "/" = {
+            root = pkgs.python3.arkheon.frontend.override {
+                backendUrl = "http${lib.optionalString cfg.reverseProxy.enableHTTPS "s"}://${}/api";
+            };
+            tryFiles = "$uri $uri/ $uri.html /index.html";
```

Normalement ça devrait suffire
```suggestion
            tryFiles = "$uri /index.html";
```